### ISSUE TITLE
Java.sql.Date.getHours throws exception at org.thymeleaf.util.JavaScriptUtils.printObject

### ISSUE THREAD

**Issue by AskerKazharov** [thymeleaf/thymeleaf#121]
We use Hibernate to DB processing. Example of date field:

```
@Temporal(javax.persistence.TemporalType.DATE)
@Column
private Date startDate;
```

startDate is mapped as java.sql.Date, which has the following deprecated method:

```
public int getHours() {
throw new java.lang.IllegalArgumentException();
}
```

So we got the following exception:

> java.lang.RuntimeException: org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.thymeleaf.exceptions.TemplateProcessingException: Error during execution of processor 'org.thymeleaf.standard.processor.text.StandardTextInliningTextProcessor'
>   at org.atmosphere.handler.ReflectorServletProcessor.onRequest(ReflectorServletProcessor.java:175)
>   at org.atmosphere.cpr.AsynchronousProcessor.action(AsynchronousProcessor.java:256)
>   at org.atmosphere.cpr.AsynchronousProcessor.suspended(AsynchronousProcessor.java:166)
>   at org.atmosphere.container.BlockingIOCometSupport.service(BlockingIOCometSupport.java:101)
>   at org.atmosphere.cpr.AtmosphereFramework.doCometSupport(AtmosphereFramework.java:1307)
>   at org.atmosphere.cpr.AtmosphereServlet.doPost(AtmosphereServlet.java:293)
>   at org.atmosphere.cpr.AtmosphereServlet.doGet(AtmosphereServlet.java:279)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:621)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
>   at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:304)
>   at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
>   at com.asaas.payroll.frontend.web.filter.CleanUrlParameterFilter.doFilter(CleanUrlParameterFilter.java:136)
>   at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:243)
>   at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
>   at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:224)
>   at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:169)
>   at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472)
>   at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:168)
>   at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:100)
>   at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:929)
>   at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
>   at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:405)
>   at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:964)
>   at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:515)
>   at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:302)
>   at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
>   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
>   at java.lang.Thread.run(Thread.java:662)
> Caused by: org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.thymeleaf.exceptions.TemplateProcessingException: Error during execution of processor 'org.thymeleaf.standard.processor.text.StandardTextInliningTextProcessor'
>   at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:894)
>   at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:778)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:621)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
>   at org.atmosphere.util.AtmosphereFilterChain.doFilter(AtmosphereFilterChain.java:171)
>   at org.atmosphere.util.AtmosphereFilterChain.invokeFilterChain(AtmosphereFilterChain.java:132)
>   at org.atmosphere.handler.ReflectorServletProcessor$FilterChainServletWrapper.service(ReflectorServletProcessor.java:314)
>   at org.atmosphere.handler.ReflectorServletProcessor.onRequest(ReflectorServletProcessor.java:172)
>   ... 27 more
> Caused by: org.thymeleaf.exceptions.TemplateProcessingException: Error during execution of processor 'org.thymeleaf.standard.processor.text.StandardTextInliningTextProcessor'
>   at org.thymeleaf.processor.AbstractProcessor.process(AbstractProcessor.java:223)
>   at org.thymeleaf.dom.Node.applyNextProcessor(Node.java:897)
>   at org.thymeleaf.dom.Node.processNode(Node.java:858)
>   at org.thymeleaf.dom.NestableNode.computeNextChild(NestableNode.java:639)
>   at org.thymeleaf.dom.NestableNode.doAdditionalProcess(NestableNode.java:621)
>   at org.thymeleaf.dom.Node.processNode(Node.java:877)
>   at org.thymeleaf.dom.NestableNode.computeNextChild(NestableNode.java:639)
>   at org.thymeleaf.dom.NestableNode.doAdditionalProcess(NestableNode.java:621)
>   at org.thymeleaf.dom.Node.processNode(Node.java:877)
>   at org.thymeleaf.dom.NestableNode.computeNextChild(NestableNode.java:639)
>   at org.thymeleaf.dom.NestableNode.doAdditionalProcess(NestableNode.java:621)
>   at org.thymeleaf.dom.Node.processNode(Node.java:877)
>   at org.thymeleaf.dom.NestableNode.computeNextChild(NestableNode.java:639)
>   at org.thymeleaf.dom.NestableNode.doAdditionalProcess(NestableNode.java:621)
>   at org.thymeleaf.dom.Node.processNode(Node.java:877)
>   at org.thymeleaf.dom.Document.process(Document.java:93)
>   at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1243)
>   at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1148)
>   at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1095)
>   at org.thymeleaf.spring3.view.ThymeleafView.renderFragment(ThymeleafView.java:221)
>   at org.thymeleaf.spring3.view.ThymeleafView.render(ThymeleafView.java:146)
>   at org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1180)
>   at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:950)
>   at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:852)
>   at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:882)
>   ... 34 more
> Caused by: java.lang.IllegalArgumentException: Could not perform introspection on object of class java.sql.Date
>   at org.thymeleaf.util.JavaScriptUtils.printObject(JavaScriptUtils.java:349)
>   at org.thymeleaf.util.JavaScriptUtils.print(JavaScriptUtils.java:170)
>   at org.thymeleaf.util.JavaScriptUtils.printKeyValue(JavaScriptUtils.java:326)
>   at org.thymeleaf.util.JavaScriptUtils.printMap(JavaScriptUtils.java:317)
>   at org.thymeleaf.util.JavaScriptUtils.printObject(JavaScriptUtils.java:345)
>   at org.thymeleaf.util.JavaScriptUtils.print(JavaScriptUtils.java:170)
>   at org.thymeleaf.util.JavaScriptUtils.printKeyValue(JavaScriptUtils.java:326)
>   at org.thymeleaf.util.JavaScriptUtils.printMap(JavaScriptUtils.java:317)
>   at org.thymeleaf.util.JavaScriptUtils.printObject(JavaScriptUtils.java:345)
>   at org.thymeleaf.util.JavaScriptUtils.print(JavaScriptUtils.java:170)
>   at org.thymeleaf.util.JavaScriptUtils.printKeyValue(JavaScriptUtils.java:326)
>   at org.thymeleaf.util.JavaScriptUtils.printMap(JavaScriptUtils.java:317)
>   at org.thymeleaf.util.JavaScriptUtils.printObject(JavaScriptUtils.java:345)
>   at org.thymeleaf.util.JavaScriptUtils.print(JavaScriptUtils.java:170)
>   at org.thymeleaf.util.JavaScriptUtils.printKeyValue(JavaScriptUtils.java:326)
>   at org.thymeleaf.util.JavaScriptUtils.printMap(JavaScriptUtils.java:317)
>   at org.thymeleaf.util.JavaScriptUtils.printObject(JavaScriptUtils.java:345)
>   at org.thymeleaf.util.JavaScriptUtils.print(JavaScriptUtils.java:170)
>   at org.thymeleaf.util.JavaScriptUtils.print(JavaScriptUtils.java:54)
>   at org.thymeleaf.standard.inliner.StandardJavaScriptTextInliner.formatEvaluationResult(StandardJavaScriptTextInliner.java:46)
>   at org.thymeleaf.standard.inliner.AbstractStandardScriptingTextInliner.processScriptingVariableInline(AbstractStandardScriptingTextInliner.java:248)
>   at org.thymeleaf.standard.inliner.AbstractStandardScriptingTextInliner.processScriptingInline(AbstractStandardScriptingTextInliner.java:85)
>   at org.thymeleaf.standard.inliner.AbstractStandardScriptingTextInliner.inline(AbstractStandardScriptingTextInliner.java:72)
>   at org.thymeleaf.standard.processor.text.StandardTextInliningTextProcessor.processTextNode(StandardTextInliningTextProcessor.java:93)
>   at org.thymeleaf.processor.text.AbstractTextNodeProcessor.doProcess(AbstractTextNodeProcessor.java:69)
>   at org.thymeleaf.processor.AbstractProcessor.process(AbstractProcessor.java:212)
>   ... 58 more
> Caused by: java.lang.reflect.InvocationTargetException
>   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
>   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
>   at java.lang.reflect.Method.invoke(Method.java:597)
>   at org.thymeleaf.util.JavaScriptUtils.printObject(JavaScriptUtils.java:340)
>   ... 83 more
> Caused by: java.lang.IllegalArgumentException
>   at java.sql.Date.getHours(Date.java:177)
>   ... 88 more

readMethod.invoke(object) throws InvocationTargetException after call deprecated getter.

We propose to process java.sql.Date as special case.

**Comment by ultraq** [comment]
Added a special case in the Javascript/Dart inline text utils for `java.sql.Date` types, converting it to a `java.util.Date` and using that object's `toString` method.  Slated for the 2.1.3 release.

**Comment by danielfernandez** [comment]
Updated 2.1.3-SNAPSHOT

**Comment by danielfernandez** [comment]
Note this fix is broken, as explained in #290
